src/maps/OverlayedMap.cpp

```diff
@@ -100,32 +101,37 @@ void OverlayedMap::centerOnWorldPos(double latitude, double longitude) {
     }
 }
 
-void OverlayedMap::setPlanePosition(double latitude, double longitude, double heading) {
+void OverlayedMap::setPlaneLocations(std::vector<avitab::Location> &locs) {
     if (!tileSource->supportsWorldCoords()) {
         return;
     }
 
-    double deltaLat = std::abs(latitude - this->planeLat);
-    double deltaLon = std::abs(longitude - this->planeLong);
-    double deltaHeading = std::abs(heading - this->planeHeading);
+    bool movement = false;
+    for (size_t i = 0; i < locs.size(); ++i) {
+        if (i < planeLocations.size()) {
+            double deltaLat = std::abs(locs[i].latitude - planeLocations[i].latitude);
+            double deltaLon = std::abs(locs[i].longitude - planeLocations[i].longitude);
+            double deltaHeading = std::abs(locs[i].heading - planeLocations[i].heading);
+            movement |= (deltaLat > 0.0000001 || deltaLon > 0.0000001 || deltaHeading > 0.1);
+        } else {
+            movement = true;
+        }
+    }
+    planeLocations = locs;
 
-    if (deltaLat > 0.0000001 || deltaLon > 0.0000001 || deltaHeading > 0.1) {
-        planeLat = latitude;
-        planeLong = longitude;
-        planeHeading = heading;
+    if (movement) {
         stitcher->updateImage();
     }
 }
 
-void OverlayedMap::centerOnPlane(double latitude, double longitude, double heading) {
+void OverlayedMap::centerOnPlane() {
     if (!tileSource->supportsWorldCoords()) {
         return;
     }
 
-    planeLat = latitude;
-    planeLong = longitude;
-    planeHeading = heading;
-    centerOnWorldPos(latitude, longitude);
+    if (planeLocations.size() > 0) {

```

Comment:
There's `vector::empty()` to check if a vector is empty, but that's just a minor thing.